### ISSUE TITLE
[Research] Tracking functionality 

### ISSUE THREAD

**Issue by toddbaert** [open-feature/ofep#16]
Many SDKs have functionality to "track" specific user actions. These typically seem to be methods that can be called to record that a certain flag-related event has taken place - often some goal related to a business outcome (though this paradigm might not be universally true for all implementations).

Examples: 

https://help.split.io/hc/en-us/articles/360020585772-Events
https://docs.launchdarkly.com/sdk/features/events#tracking-events

It should be noted that some vendors don't have the functionality, so we will want to consider that in our implementation. If we decide to make this sort of tracking the business of OpenFeature, we may want to allow for registering a separate tracking provider in such cases.

cc @weyert 

Relates to: https://github.com/open-feature/spec/pull/102

**Comment by weyert** [comment]
PostHog supports the following:

```javascript
posthog.capture('[event-name]', {property1: 'value', property2: 'another value'});
```

**Comment by weyert** [comment]
FlagShip.io appears to support some sort of track events via:

```javascript
import { HitType, EventCategory } from "@flagship.io/js-sdk";

visitor.sendHit({
  type: HitType.EVENT, //or "EVENT"
  category: EventCategory.USER_ENGAGEMENT, // or EventCategory.ACTION_TRACKING
  action: "click",
  label: "label",
  value: 100,
});
```

See: https://docs.developers.flagship.io/docs/js-v3#hit-tracking

**Comment by beeme1mr** [comment]
Hey @weyert, would you be willing to put together an [OFEP](https://github.com/open-feature/research/blob/main/000-OFEP-template.md) for adding tracking events to the spec?

**Comment by justinabrahms** [comment]
I think this is the same as https://github.com/open-feature/spec/issues/78  ?

**Comment by beeme1mr** [comment]
Yes it's the same thing. I'll close the one in the spec in favor of the research issue. The spec can be updated once an OpenFeature Enhancement Proposal has been accepted.

**Comment by InTheCloudDan** [comment]
I came across this repo: https://github.com/davidwells/analytics which is for a similar use case it seems and might be able to provide inspiration.

**Comment by powelli** [comment]
Not sure about other providers, but with Optimizely at least a tracking event/impression can be recorded each time a flag evaluation is carried out: https://support.optimizely.com/hc/en-us/articles/4410284241165#Full_Stack

The firing of these events can be controlled as part of the client SDK initialisation, and via additional context passed to the evaluation method: https://docs.developers.optimizely.com/full-stack/v4.0/docs/decide-methods-java#optimizelydecideoption

This is useful in some use-cases we have as costs are based on the volume of tracked events, but some evaluations don't need to be tracked (e.g. if they're not associated with an experiment).

**Comment by justinabrahms** [comment]
@s-sen - This may be an issue y'all would like to get involved with.

**Comment by weyert** [comment]
I am going to pick up this issue again. Interested to sort something out.

**Comment by jrydberg** [comment]
A few vendors (segment.io, Rudderstack, Hightouch) have standardized on this specification:

* https://segment.com/docs/connections/spec/

One good option would be to adopt and make an open version of the spec. Either as first-class citizen in OpenFeature or as a sister project.

**Comment by tapico-weyert** [comment]
What do you mean with 'One good option would be to adopt and make an open version of the spec.'? 

Do you mean a OpenFeature spec or more separate from OpenFeature? Or its discussed in sibling spec in the OpenFeature repo?

**Comment by weyert** [comment]
The standard looks alright but I think it has a bit awkward wording (e.g. `traits`) in the context of OpenFeature maybe we can adopt the idea but change the schema.

**Comment by beeme1mr** [comment]
Tracking as been added to the spec.
